src/syncMacros.ts

```diff
@@ -46,7 +46,7 @@ export async function syncMacros(hackmudPath: string) {
 	}
 
 	for (const user of users)
-		writeFile(resolvePath(hackmudPath, `${user}.macros`), macroFile)
+		await writeFile(resolvePath(hackmudPath, `${user}.macros`), macroFile)
```

Comment:
oh I just figured out why this is happening. this used to work but one day I added `process.exit()` to `src/bin/hsm.js` to exit immediately regardless of unfinished promises.

it's generally a bad idea to have a loop with `await` since you're waiting to start doing the async thing only once the previous async thing has finished.

this should be replaced with `await Promise.all(users.map(async user => writeFile(resolvePath(hackmudPath, `${user}.macros`), macroFile)))`

Reply:
I only checked documentation what's possible after I saw that this command clear macros, if you have better way to solve it, great :)